### ISSUE TITLE
brush machine offline

### ISSUE THREAD

**Issue by rradar** [itead/Sonoff_Devices_DIY_Tools#33]
How to do it @ZZLinvec ? I'm on a linux host!

**Comment by ZZLinvec** [comment]
We currently have no intention of developing a Linux version of the tool.Now I say my code running process, you can follow my code, some windos libraries into Linux on the OK.
Flash the firmware dialog code in Dialog_text.py》class RootDialog.
First, get the IP address and port number of the PC.
Second, set up an HTTP server,
Third, send unlock information and server information to the device.
In the case of these steps, the substitution that should be required is step one.
The corresponding code is located near 618 of the file dialog_text.py

**Comment by alexbk66** [comment]
I asked in another issue - what's the procedure to update firmware to latest version?  Do I need to download bin file and do all these steps manually?

**Comment by alexbk66** [comment]
I'm writing a C# app to replace the Python tool which should work on Linux (using mono)

**Comment by rradar** [comment]
And how does it work offline @ZZLinvec ? I don't want contact to cloud!

**Comment by ZZLinvec** [comment]
Flash firmware here refers to downloading third-party firmware to the device.Note that this would mean breaking the firmware of our device.We will no longer guarantee it.

**Comment by ZZLinvec** [comment]
What does offline mean?
Devices downloading firmware in DIY mode must be reported to the server, so the unlocking step cannot be skipped

**Comment by alexbk66** [comment]
> downloading third-party firmware to the device

Why anybody will need to bother with DIY version to "downloading third-party firmware to the device"?

The whole idea of DYI devices is to use them without flashing the FW.  

The only need to flash new FW- is to upgrade FW to latest Sonoff DIY version **without** going back to "non-DIY" mode and using ewelink app to update FW.  Then going through all the pain of changing back to DIY mode.

**Comment by ZZLinvec** [comment]
@alexbk66  No, do it yourself. This firmware is not just a way to use DIY, it's more important to develop FW yourself. This is no longer just DIY control under the LAN.

**Comment by alexbk66** [comment]
> @alexbk66 No, do it yourself. This firmware is not just a way to use DIY, it's more important to develop FW yourself. This is no longer just DIY control under the LAN.

What do you mean "do it yourself"?  So you suggest switching to eWeLink mode to update the firmware and then go through all the pain of changing to DIY mode again?

Seriously @ZZLinvec - you have to start thinking more about your (potential) customers!
If you don't - I will tell all my users (HomeSeer) to forget about Sonoff and choose Shelly devices - they are very professionally implemented.
And I will make sure your management is also informed. It's really getting onto my nerves, I wasted so much time.

**Comment by ZZLinvec** [comment]
This is a misunderstanding, I mean we have more DIY space, not to ignore your opinion

**Comment by alexbk66** [comment]
Yeah?  I asked many times - how to upgrade **Sonoff** firmware?
You agree that **manually** upgrading 20 Sonoff devices in my house is simply not feasible.

**Comment by alexbk66** [comment]
> This is a misunderstanding, I mean we have more DIY space, not to ignore your opinion

Most enthusiasts who want to flash **third party firmware** don't mind initially wiring the device to the computer.  The best advantage of DIY devices is **local control** - that's all most users need, so third party firmware is not really needed anymore.

**Comment by alexbk66** [comment]
@ZZLinvec did you look at Shelly devices? They support without any jumpers/flashing:
1. WiFi Access Point (AP mode) or 
2. Client Mode (STA)
3. MQQT
4. REST
5. CoIoT commands
6. Cloud control
7. HTTP Webserver

**Comment by alexbk66** [comment]
https://www.youtube.com/watch?v=zFIPrJybt94

**Comment by JAMESBOWLER** [comment]
> I asked in another issue - what's the procedure to update firmware to latest version? Do I need to download bin file and do all these steps manually?

To update offical firmware to device it needs to do it through ewilink app

**Comment by alexbk66** [comment]
> To update offical firmware to device it needs to do it through ewilink app

I guess so. But that's where the problem is - I have to switch to eWiLink mode, update FW, then go through the pain of changing back to DIY mode again?  And that's for my 20 devices?

**Comment by ZZLinvec** [comment]
@alexbk66 Ok, I understand your needs. I want to know more about your needs. Are there any other ways of instant messaging?It is more convenient for us to communicate and improve our efficiency.

**Comment by alexbk66** [comment]
> @alexbk66 Ok, I understand your needs. I want to know more about your needs. Are there any other ways of instant messaging?It is more convenient for us to communicate and improve our efficiency.

Sure - Skype, Viber, Whatsapp

**Comment by alexbk66** [comment]
Hello @ZZLinvec

**Comment by rradar** [comment]
@ZZLinvec :grey_question:

**Comment by JAMESBOWLER** [comment]
@rradar   (I don't work for Itead)
Have you used the ewilink app to update to the latest firmware? This is the only way supported by itead
After that you can use a program like postman to send the json request to flash your bin file.

**Comment by alexbk66** [comment]
> @alexbk66 Ok, I understand your needs. I want to know more about your needs. Are there any other ways of instant messaging?It is more convenient for us to communicate and improve our efficiency.

So, @ZZLinvec - Skype, Viber, Whatsapp?

**Comment by rradar** [comment]
I see itead is not taking things seriously. I thought I could deploy the sonoff mini's directly in new constructions (talking about 100/1000+ devices) but if the software support is so miserable I better switch to a serious vendor like using the shelly.

Nice try @iteadsw @ZZLinvec but good hardware alone isn't worth it! If you don't let the users decide if they want to use your sh!tty cloud/apps etc. you are out of business...

**Comment by alexbk66** [comment]
> I see itead is not taking things seriously. I thought I could deploy the sonoff mini's directly in new constructions (talking about 100/1000+ devices) but if the software support is so miserable I better switch to a serious vendor like using the shelly.
> 
> Nice try @iteadsw @ZZLinvec but good hardware alone isn't worth it! If you don't let the users decide if they want to use your sh!tty cloud/apps etc. you are out of business...

Well put!  
Shelly actually released a dimmer device!
